docs/user-guide/SPIN.rst

```diff
@@ -0,0 +1,181 @@
+               ++model.micro_batch_size=1 \
+               ++model.global_batch_size=64 \
+               exp_manager.explicit_log_dir=${RESULTS_DIR} \
+               exp_manager.create_wandb_logger=True \
+               exp_manager.wandb_logger_kwargs.name=${NAME} \
+               exp_manager.wandb_logger_kwargs.project=${PROJECT} \
+               ++trainer.spin.max_iterations=1 \
+               ++trainer.spin.max_epochs=1 \
+               ++model.spin.ref_policy_kl_penalty=0.1
+               ++model.spin.length_params.max_length=2048 \
+               ++model.data.train_ds.max_seq_length=4096 \
+            EOF
+
+            srun -o $OUTFILE -e $ERRFILE --container-image=$CONTAINER $MOUNTS bash -c "${cmd}"
+            set +x
+
+During SPIN training, there will be several metrics recorded to WandB which you can monitor, chiefly acc (representing the percentage amount whereby the model's actual [ground truth label] rewards are greater than the generated rewards [generated by the reference policy]).
+The ``reward`` in this case is calculated as the difference between model log probs and the reference log probs, multiplied by the KL penalty (beta in the original paper), for the ground truth and generated responses.
+During training, the acc should generally be increasing, but don't worry if its absolute value remains low, as it doesn't correlate to finalised MTBench or MMLU scores. It should just be generally increasing.
+Other metrics to keep an eye on are the rewards_actual_mean and rewards_generated_mean, which represent the average of the ``rewards`` as defined above. Again, the absolute values aren't necessarily so important as the fact that the actual_mean should be greater than the generated_mean over time, and the greater that difference, the better.
+All metrics will be grouped by either ``train/`` or ``val/`` in WandB, representing whether that metric is from the training or validation set, respectively.
+
+When it comes to ideal hyperparameters for SPIN training, much will depend on the characteristics of your SFT (or base/foundational) model and your training data, so there is no one-size-fits-all parameter set which will work in all cases.
+However, the following the following is a brief overview of which hyperparameters we have perturbed for various model sizes and their effects:
+
+* global_batch_size: the SPIN paper recommends 64 for a 7B model, which we have found holds true, in that higher GBS for 7B models performs much worse. For larger models, you can increase to 128 or 256 as needed, but recommend you start with 64 as a baseline
+* iterations/epochs: the SPIN paper used iterations=3 and epochs=2 for their training on a 7B model with a training dataset size of 200k. Using the same foundational model as the authors, we found better results with iterations=1, epochs=1 using a 50k subset of their 200k data. We therefore recommend starting with iterations=1, and increasing to 2 as needed by testing on MT-Bench/MMLU.
```

Comment:
I think we should mention the fact that our current implementation doesn't increase the size of the training set on iterations 2+, as in the SPIN paper. We don't know to which extent this may be important to get good results with more iterations (right?).

Reply:
I have added a line about that, let me know if it suffices